kubectl-plugin/pkg/cmd/cluster/cluster_session.go

```diff
@@ -0,0 +1,110 @@
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/kubectl/pkg/cmd/portforward"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+type ClusterSessionOptions struct {
+	ioStreams    *genericiooptions.IOStreams
+	configFlags  *genericclioptions.ConfigFlags
+	ResourceName string
+	Namespace    string
+}
+
+func NewClusterSessionOptions(streams genericiooptions.IOStreams) *ClusterSessionOptions {
+	return &ClusterSessionOptions{
+		ioStreams:   &streams,
+		configFlags: genericclioptions.NewConfigFlags(true),
+	}
+}
+
+func NewClusterSessionCommand(streams genericiooptions.IOStreams) *cobra.Command {
+	options := NewClusterSessionOptions(streams)
+	factory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:   "session NAME",
+		Short: "Forward local ports to the RayCluster.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), factory)
+		},
+	}
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *ClusterSessionOptions) Complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+	options.ResourceName = args[0]
+
+	if *options.configFlags.Namespace == "" {
+		options.Namespace = "default"
+	} else {
+		options.Namespace = *options.configFlags.Namespace
+	}
+
+	return nil
+}
+
+func (options *ClusterSessionOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+	return nil
+}
+
+func (options *ClusterSessionOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	kubeClientSet, err := factory.KubernetesClientSet()
+	if err != nil {
+		return fmt.Errorf("dynamic client failed to initialize: %w", err)
+	}
+
+	listopts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("ray.io/cluster=%s, ray.io/node-type=head", options.ResourceName),
+	}
+
+	rayHeadSvcs, err := kubeClientSet.CoreV1().Services(options.Namespace).List(ctx, listopts)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve ray head services: %w", err)
+	}
+
+	if len(rayHeadSvcs.Items) == 0 {
+		return fmt.Errorf("no ray head services found")
+	}
+	if len(rayHeadSvcs.Items) > 1 {
+		return fmt.Errorf("more than one ray head service found")
+	}
+
+	rayHeadSrc := rayHeadSvcs.Items[0]
+
+	portForwardCmd := portforward.NewCmdPortForward(factory, *options.ioStreams)
+	portForwardCmd.SetArgs([]string{"service/" + rayHeadSrc.Name, "0:8265"})
```

Comment:
Another possibility instead of using `0:8265` is to default the local port to 8265 and make it configurable?